### ISSUE TITLE
Use of hanging comma in index expression should result in an error

### ISSUE THREAD

**Issue by erictraut** [microsoft/pyright#1444]
The following code emits a runtime error because `0,` is interpreted as `(0,)`, yet pyright doesn't flag it as an error.

```python
x = [3, 4, 5]

x[0,] # List indices must be integers or slices, not tuple
```

**Comment by erictraut** [comment]
This is included in Pyright 1.1.108, which I just published. It will also be in the next release of Pylance.
